### PR TITLE
Add new dose release (4.1)

### PR DESCRIPTION
**Pull request by abate**
new upstream release

### CODE REVIEW

**Comment by dsheets** [comment]
It looks like there is an issue with module aliases on older compilers and a potential issue with cppo/OCaml 4.03.0.

**Comment by abate** [comment]
all clean !

**Comment by dsheets** [comment]
I get

```
File "applications/distcheck.ml", line 28, characters 33-65:
Error: Signature mismatch:
       ...
       Values do not match:
         val options :
           ?usage:string ->
           ?version:string ->
           ?suppress_usage:bool ->
           ?suppress_help:bool ->
           ?prog:string ->
           ?formatter:OptParse.Formatter.t -> unit -> OptParse.OptParser.t
       is not included in
         val options :
           ?usage:string ->
           ?status:int ->
           ?version:string ->
           ?suppress_usage:bool ->
           ?suppress_help:bool ->
           ?prog:string ->
           ?formatter:OptParse.Formatter.t -> unit -> OptParse.OptParser.t
       File "applications/distcheck.ml", line 28, characters 44-51:
         Actual declaration
Command exited with code 2.
```

on OCaml 4.02.3 on OS X 10.11.2. I don't know how that passed CI but this breaks `opam-publish` for me.

File: packages/dose/dose.3.4.1/opam [hunk 1]
```diff
@@ -15,20 +15,14 @@ bug-reports: "https://gforge.inria.fr/tracker/?group_id=4395"
 license: "LGPL-v3+ with OCaml linking exception"
 dev-repo: "https://gforge.inria.fr/git/dose/dose.git"
 build: [
-  ["./configure" "--with-ocamlgraph" "--bindir=%{bin}%" "--prefix=%{prefix}%"]
+  ["./configure" "--with-ocamlgraph"]
+  [make printconf]
```
Comment (dsheets) [inline packages/dose/dose.3.4.1/opam]: I think `printconf` should be `"printconf"`?

**Comment by abate** [comment]
For the compilation problem, it probably passed the CI because I compile only the library and not the applications. Do you need the application distcheck for opam-publish ?

File: packages/dose/dose.3.4.1/opam (hunk 1 above)
Comment (abate) [inline packages/dose/dose.3.4.1/opam]: I can suppress this altogether. It was there mostly for debugging ...

**Comment by yallop** [comment]
@dsheets: https://github.com/ocaml/opam-repository/pull/5399

**Pull request merged**